server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -2791,6 +2792,16 @@ public void testCheckpointOnDiskFull() throws IOException {
         assertEquals(read, checkpoint);
     }
 
+    public void testLegacyCheckpointVersion() throws IOException {
+        expectThrows(
+            TranslogCorruptedException.class,
+            IndexFormatTooOldException.class,
+            () -> Checkpoint.read(getDataPath("/org/elasticsearch/index/checkpoint/v1.cpk"))
+        );
+        // just check that it doesn't fail
+        Checkpoint.read(getDataPath("/org/elasticsearch/index/checkpoint/v2.cpk"));
```

Comment:
While we're here let's assert it reads it right and fills in the missing value with `SequenceNumbers.UNASSIGNED_SEQ_NO` rather than just that it doesn't fail:

```suggestion
        assertThat(Checkpoint.read(getDataPath("/org/elasticsearch/index/checkpoint/v2.ckp")), equalTo(new Checkpoint(-1312746831014894010L,
            44230819, 4168771208509507653L, 6217263213205155568L, 8590850694628654668L, 3768575734506660560L, 1476009383806516272L,
            SequenceNumbers.UNASSIGNED_SEQ_NO)));
```